types/async/index.d.ts

```diff
@@ -341,14 +342,14 @@ export function groupByLimit<T, K, E = Error>(iterable: IterableCollection<T>, l
 export const groupBySeries: typeof groupBy;
 
 // Control Flow
-export function series<T, E = Error>(tasks: Array<AsyncFunction<T, E>>, callback?: AsyncResultArrayCallback<T, E>): void;
-export function series<T, E = Error>(tasks: Dictionary<AsyncFunction<T, E>>, callback?: AsyncResultObjectCallback<T, E>): void;
+export function series<T, E = Error>(tasks: Array<AsyncFunction<T, E>>, callback: AsyncResultArrayCallback<T, E>): void;
+export function series<T, E = Error>(tasks: Dictionary<AsyncFunction<T, E>>, callback: AsyncResultObjectCallback<T, E>): void;
 export function series<T, R, E = Error>(tasks: Array<AsyncFunction<T, E>> | Dictionary<AsyncFunction<T, E>>): Promise<R>;
```

Comment:
Ideally this type would be refactored into two functions, one for an array and one for a dictionary, just like the the callback variants above. This would render the second `R` generic type unnecessary, as we'd know that it will either return `T[]` or `Dictionary<T>`. However, that would be a breaking change, so I didn't pursue that.